pkgs/tools/archivers/peazip/default.nix

```diff
@@ -0,0 +1,93 @@
+{ mkDerivation
+, lib
+, fetchFromGitHub
+, wrapQtAppsHook
+, fpc
+, lazarus
+, xorg
+, libqt5pas
+, _7zz
+, archiver
+, brotli
+, upx
+, zpaq
+, zstd
+}:
+
+mkDerivation rec {
+  pname = "peazip";
+  version = "8.6.0";
+
+  src = fetchFromGitHub {
+    owner = "peazip";
+    repo = pname;
+    rev = version;
+    sha256 = "sha256-MF0MwIKV/25lm+IKRdFXSMiXSQxTKksGBj/r0td+v/o=";
+  };
+  sourceRoot = "source/peazip-sources";
```

Comment:
```suggestion
  sourceRoot = "${src.name}/peazip-sources";
```

Reply:
There is no `name` in `src`.

Reply:
There is https://github.com/NixOS/nixpkgs/blob/1540d4fd8dabdf04de8280838ecc387051ccd8f4/pkgs/build-support/fetchgithub/default.nix#L3

Reply:
But why should we use that? The source is still in `source` not in `name`. Or is it there too?

Reply:
"source" is just default value for `src.name`, but someone can override `src` with some custom `name`, in such case `sourceRoot` will be broken.

Reply:
We are not overriding it here tho.

Reply:
And if you override `src.name` then why is `src.name` still the thing we want?

Reply:
Well, what about such example:
```nix
nixpkgs.overlays = [
  (self: super: {
    peazip = super.peazip.overrideAttrs (oldAttrs: {
      src = super.fetchFromGitHub {
        name = "foobar";
        ...
      };
    });
  })
];
```